src/App.js

```diff
@@ -1,85 +1,273 @@
+
+    this.setState({
+      task: value.replace(/^\s+$/g, ''),
+    });
+  }
+
+  addTodo = (event) => {
+    event.preventDefault();
+    const { task } = this.state;
+
+    this.setState(prevState => ({
+      todos: [...prevState.todos, {
+        id: uuidv4(), task, completed: false,
+      }],
+      task: '',
+      isAnyFiltered: false,
+      filtered: [],
+      placeholder: 'What needs to be done?',
+      activeLink: 'All',
+    }));
+  }
+
+  changeStatus = (id, status) => {
+    const { todos, filtered } = this.state;
+    let newListFiltered;
+
+    const newList = todos.map((item) => {
+      const newItem = { ...item };
+
+      if (newItem.id === id) {
+        newItem.completed = status;
+      }
+
+      return newItem;
+    });
+
+    if (filtered) {
+      newListFiltered = filtered.map((item) => {
+        const newItem = { ...item };
+
+        if (newItem.id === id) {
+          newItem.completed = status;
+        }
+
+        return newItem;
+      });
+    }
+
+    this.setState({
+      todos: [...newList],
+      filtered: newListFiltered,
+      isCheckedAll: todos.every(item => item.completed === true),
+    });
+  }
+
+  removeTask = (name) => {
+    const { todos, filtered } = this.state;
+    const newList = todos.filter(item => item.id !== name);
+    let newListFiltered;
+
+    if (filtered) {
+      newListFiltered = filtered.filter(item => item.id !== name);
+    } else {
+      newListFiltered = filtered;
+    }
+
+    this.setState({
+      todos: [...newList],
+      filtered: newListFiltered,
+    });
+  }
+
+  removeCompleted = () => {
+    const { todos, filtered } = this.state;
+    const newList = todos.filter(item => item.completed === false);
+    let newListFiltered;
+
+    if (filtered) {
+      newListFiltered = filtered.filter(item => item.completed === false);
+    }
+
+    this.setState({
+      todos: [...newList],
+      filtered: newListFiltered,
+      isCheckedAll: false,
+    });
+  }
+
+  showAll = (event) => {
+    event.preventDefault();
+    const { innerText } = event.target;
+
+    this.setState({
+      filtered: [],
+      isAnyFiltered: false,
+      activeLink: innerText,
+    });
+  }
+
+  showCompleted = (event) => {
+    event.preventDefault();
+    const { innerText } = event.target;
+    const { todos } = this.state;
+    const newList = todos.filter(item => item.completed === true);
+
+    this.setState({
+      filtered: [...newList],
+      isAnyFiltered: true,
+      activeLink: innerText,
+    });
+  }
+
+  showActive = (event) => {
+    event.preventDefault();
+    const { innerText } = event.target;
+    const { todos } = this.state;
+    const newList = todos.filter(item => item.completed === false);
+
+    this.setState({
+      filtered: [...newList],
+      isAnyFiltered: true,
+      activeLink: innerText,
+    });
+  }
+
+  validatedForm = (event) => {
+    event.preventDefault();
+
+    this.setState({
+      placeholder: 'Please enter the task',
```

Comment:
why do you do this here? do you need to change placeholder somewhere?

Reply:
delete placeholder